pylint_odoo/checkers/modules_odoo.py

```diff
@@ -469,23 +469,32 @@ def _check_redundant_modulename_xml(self):
             return False
         return True
 
-    def _get_duplicate_xml_fields(self, fields):
+    def _get_fields_by_parents(self, fields):
+        parents = {}
+        all_fields = {}
+        for field in fields:
+            field_xml = field.attrib.get('name')
+            if not field_xml:
+                continue
+            parent = field.getparent()
+            parents.setdefault(
+                parent, {}).setdefault((field_xml, parent), []).append(field)
+            all_fields.setdefault((field_xml, parent), []).append(field)
+        return parents, all_fields
```

Comment:
You don't need 2 dictionaries to detect duplicated.
I mean, you could use just 1 dictionary since that you have the complex key:
 - field_xml <- str
 - parent <- node object (this is different although name string of parent is equal)

And the appended values are fields
Then you could detect them with
`[(field_xml, nodes) for (field_xml, parent), nodes in all_fields.items() if len(nodes) >= 2)`